File: 3513-number-of-unique-xor-triplets-i/3513-number-of-unique-xor-triplets-i.cpp
class Solution {
public:
    int uniqueXorTriplets(vector<int>& nums) {
        // Get the total number of elements in the input vector
        int n = nums.size();
        
        // Base case: If there is only 1 element, it can form 1 unique configuration
        if(n == 1) return 1;
        
        // Base case: If there are 2 elements, they can form 2 unique configurations
        if(n == 2) return 2;
        
        int temp;
        int i = 0;
        
        // Find the smallest power of 2 that is strictly greater than n
        while(n >= temp) { 
            // Calculate 2 raised to the power of i (2^i)
            temp = pow(2, i); 
            
            // Increment the exponent for the next iteration
            i++; 
        }
        
        // Return the calculated power of 2
        return temp; 
    }
};

File: 3513-number-of-unique-xor-triplets-i/3513-number-of-unique-xor-triplets-i.java
class Solution {
    public int uniqueXorTriplets(int[] nums) {
        int n = nums.length;
        if(n==1) return 1;
        if(n==2) return 2;

        int temp=0;
        int i =0;
        while(n>=temp){
            temp = (int) Math.pow(2,i);
            i++;
        }
        return temp;
    }
}